dartagnan/src/main/java/com/dat3m/dartagnan/wmm/relation/binary/RelIntersection.java

```diff
@@ -63,6 +65,30 @@ public TupleSet getMaxTupleSetRecursive(){
         return getMaxTupleSet();
     }
 
+	@Override
+	public boolean disable(TupleSet t) {
+		super.disable(t);
+		if(t.isEmpty())
+			return false;
+		TupleSet t1 = new TupleSet();
+		t1.addAll(t);
+		t1.retainAll(r2.getMinTupleSet());
+		boolean b = r1.disable(t1);
+		TupleSet t2 = new TupleSet();
+		t2.addAll(t);
+		t2.retainAll(r1.getMinTupleSet());
+		return r2.disable(t2) || b;
+	}
+
+	@Override
+	public void initEncodeTupleSet() {
+		TupleSet set = new TupleSet();
+		set.addAll(Sets.difference(disableTupleSet,Sets.intersection(r1.getDisableTupleSet(),r2.getDisableTupleSet())));
```

Comment:
I think `Sets.union` should be applicable here because as soon as a tuple is disabled in either of the sub-relations, it is trivially disabled in the intersection.